Source/JavaScriptCore/wasm/js/JSToWasm.cpp

```diff
@@ -335,16 +335,16 @@ std::unique_ptr<InternalFunction> createJSToWasmWrapper(CCallHelpers& jit, const
                     jit.storeValue(scratchJSR, addr);
                 }
             } else {
-                if (type.isF32())
-                    jit.loadFloat(jsParam, wasmFrameConvention.params[i].fpr());
-                else if (type.isF64())
-                    jit.loadDouble(jsParam, wasmFrameConvention.params[i].fpr());
-                else if (type.isI32()) {
+                if (type.isI32()) {
                     jit.load32(jsParam, wasmFrameConvention.params[i].jsr().payloadGPR());
 #if USE(JSVALUE32_64)
                     jit.move(CCallHelpers::TrustedImm32(0), wasmFrameConvention.params[i].jsr().tagGPR());
 #endif
-                } else
+                } else if (type.isF32())
+                    jit.loadFloat(jsParam, wasmFrameConvention.params[i].fpr());
+                else if (type.isF64())
+                    jit.loadDouble(jsParam, wasmFrameConvention.params[i].fpr());
+                else
                     jit.loadValue(jsParam, wasmFrameConvention.params[i].jsr());
```

Comment:
Is this just cosmetic, or are there cases where the object can be of multiple types and it's a matter of precedence?

Reply:
There are cases in which the object can be a 32-bit integer but interpreted as a 32-bit float. I suspect this is because in JavaScript, all numbers are floats, so technically a 32-bit JavaScript integer is indeed a float. Fortunately, with this reordering the floats and integers both get understood properly.

I don't know that this was causing any actual issues, but better safe than sorry, I think.